server/src/main/java/io/seata/server/session/SessionHelper.java

```diff
@@ -164,26 +164,27 @@ public static void endCommitFailed(GlobalSession globalSession, boolean retryGlo
     public static void endRollbacked(GlobalSession globalSession, boolean retryGlobal) throws TransactionException {
         if (retryGlobal || !DELAY_HANDLE_SESSION) {
             long beginTime = System.currentTimeMillis();
+            boolean timeoutDone = false;
             GlobalStatus currentStatus = globalSession.getStatus();
+            if (currentStatus == GlobalStatus.TimeoutRollbacking) {
+                MetricsPublisher.postSessionDoneEvent(globalSession, GlobalStatus.TimeoutRollbacked, false, false);
+                timeoutDone = true;
+            }
             boolean retryBranch =
-                currentStatus == GlobalStatus.TimeoutRollbackRetrying || currentStatus == GlobalStatus.RollbackRetrying;
+                    currentStatus == GlobalStatus.TimeoutRollbackRetrying || currentStatus == GlobalStatus.RollbackRetrying;
             if (SessionStatusValidator.isTimeoutGlobalStatus(currentStatus)) {
                 globalSession.changeGlobalStatus(GlobalStatus.TimeoutRollbacked);
             } else {
                 globalSession.changeGlobalStatus(GlobalStatus.Rollbacked);
             }
             globalSession.end();
-            if (!DELAY_HANDLE_SESSION) {
+            if (!DELAY_HANDLE_SESSION && !timeoutDone) {
                 MetricsPublisher.postSessionDoneEvent(globalSession, false, false);
             }
             MetricsPublisher.postSessionDoneEvent(globalSession, IdConstants.STATUS_VALUE_AFTER_ROLLBACKED_KEY, true,
-                beginTime, retryBranch);
+                    beginTime, retryBranch);
         } else {
-            if (SessionStatusValidator.isTimeoutGlobalStatus(globalSession.getStatus())) {
-                MetricsPublisher.postSessionDoneEvent(globalSession, GlobalStatus.TimeoutRollbacked, false, false);
-            } else {
-                MetricsPublisher.postSessionDoneEvent(globalSession, GlobalStatus.Rollbacked, false, false);
-            }
+            MetricsPublisher.postSessionDoneEvent(globalSession, GlobalStatus.Rollbacked, false, false);
```

Comment:
comment: retryGlobal  of TimeoutXXX must be true.
if status=TimeoutRollbacked  there is no need to print the metric again
if status=TimeoutRollbackFailed  will call endRollbackFailed
if status=TimeoutRollbackRetrying   will step into  https://github.com/seata/seata/pull/4780/files#diff-9a05c8e25f5cbde274749846da97233df30a162a9dd93078d4e41738a948a09eR165